nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncryptAttributes.java

```diff
@@ -0,0 +1,508 @@
+            .defaultValue(EncryptProcessorUtils.buildDefaultWeakCryptoAllowableValue().getValue())
+            .build();
+    public static final PropertyDescriptor ATTR_SELECT_REG_EX = new PropertyDescriptor.Builder()
+            .name("attribute-select-regex")
+            .displayName("Attributes Selection RegEx")
+            .description("If " + CUSTOM_ATTR_ALLOWABLE_VALUE.getDisplayName() + " is selcted then provied a RegEx to select " +
+                    "attributes matching a specific pattern. Only those attributes will be encrypted/decrypted " +
+                    "who matches against given regex pattern")
+            .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .defaultValue(".*")
+            .required(false)
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
+            .description("Any FlowFile that is successfully encrypted or decrypted will be routed to success").build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+            .description("Any FlowFile that cannot be encrypted or decrypted will be routed to failure").build();
+    private List<PropertyDescriptor> properties;
+
+    private Set<Relationship> relationships;
+
+    private volatile Map<String, PropertyValue> propMap = new HashMap<>();
+
+    static {
+        // add BouncyCastle encryption providers
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(ATTRS_TO_ENCRYPT);
+        properties.add(MODE);
+        properties.add(KEY_DERIVATION_FUNCTION);
+        properties.add(ENCRYPTION_ALGORITHM);
+        properties.add(ALLOW_WEAK_CRYPTO);
+        properties.add(PASSWORD);
+        properties.add(RAW_KEY_HEX);
+        properties.add(PUBLIC_KEYRING);
+        properties.add(PUBLIC_KEY_USERID);
+        properties.add(PRIVATE_KEYRING);
+        properties.add(PRIVATE_KEYRING_PASSPHRASE);
+        properties.add(ATTR_SELECT_REG_EX);
+        this.properties = Collections.unmodifiableList(properties);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+
+    @Override
+    protected Collection<ValidationResult> customValidate(final ValidationContext context) {
```

Comment:
This method is duplicated from `EncryptContent` -- probably a candidate for refactoring to `EncryptProcessorUtils`.